Description: work-around to skip a test in case we are behind a broken DNS
 Checks a domain that's supposed to be inexistent and skips a test that
 fails in case a broken DNS tries to be helpful and resolves such an
 inexistent name.
Bug-Debian: http://bugs.debian.org/671080
Origin: upstream, commit: 7b87eec0dc2298a18531b95c1cd2d1b72986e71c
--- a/test/func/netsync_badhost_gives_nice_error/__driver__.lua
+++ b/test/func/netsync_badhost_gives_nice_error/__driver__.lua
@@ -1,3 +1,22 @@
+skip_if(not existsonpath("host"))
+
+-- We punt in case of misconfigured DNS servers that resolve inexistant
+-- domain names. (Don't even think about buying that domain name!)
+L("\nChecking DNS resolution for nosuchhost__blahblah__asdvasoih.com: ")
+local pid = spawn_redirected("", "host-lookup.out", "host-lookup.err",
+                             "host", "nosuchhost__blahblah__asdvasoih.com")
+local ec = wait(pid)
+
+if ec == 0 then
+  L("failed\n",
+    "\n",
+    "Your DNS resolver is trying to be helpful by resolving names that do\n",
+    "not exist. `host nosuchhost__blahblah__asdvasoih.com` returned:\n\n")
+  log_file_contents("host-lookup.out")
+  skip_if(true)
+else
+  L("good\n")
+end
 
 mtn_setup()
 
